components/carousel/index.tsx

```diff
@@ -129,12 +131,13 @@ export default class Carousel extends React.Component<CarouselProps, {}> {
     if (props.effect === 'fade') {
       props.fade = true;
     }
-
     let className = getPrefixCls('carousel', props.prefixCls);
     const dotsClass = 'slick-dots';
     const dotPosition = this.getDotPosition();
     props.vertical = dotPosition === 'left' || dotPosition === 'right';
-    props.dotsClass = `${dotsClass} ${dotsClass}-${dotPosition || 'bottom'}`;
+    props.dotsClass = `${dotsClass} ${dotsClass}-${dotPosition || 'bottom'} ${
+      this.props.dotsClass
+    }`;
```

Comment:
Try `classNames`。

Reply:
classNames is not any prop then how it will work?

Reply:
I mean https://www.npmjs.com/package/classnames

Reply:
did u get the issue?

Reply:
the master branch has some conflicts.

Reply:
https://github.com/ant-design/ant-design/pull/21141